packages/core/src/components/inputs/SelectInput/SelectInput.vue

```diff
@@ -91,15 +88,38 @@ onMounted(() => {
       },
       {}
     )
-    selectedOption.value = optionsMap[modelValue]
+    selectedOption.value = optionsMap[modelValue.value]
   } else {
     selectedOption.value = defaultOption
   }
   if (selectedOption.value !== undefined) {
-    emitKey(selectedOption.value)
+    updateModelValue(selectedOption.value)
   }
 })
 
+watch(
+  () => modelValue.value,
+  (newValue) => {
+    if (
+      newValue !== undefined &&
+      newValue !== null &&
+      props.options.length > 0
+    ) {
+      const optionsMap = props.options.reduce(
+        (
+          accumulator: { [key: string]: SelectOption<LabelType> },
+          selectOption: SelectOption<LabelType>
+        ) => {
+          accumulator[selectOption.key || ''] = selectOption
+          return accumulator
+        },
+        {}
+      )
+      selectedOption.value = optionsMap[newValue]
+    }
+  }
+)
+
```

Comment:
Why do we need a watcher here?

Reply:
Because the modelValue was sent as a prop and the selectedOption was only configured onMounted. Thus passing the modelValue later on was not being reflected.